### PR TITLE
Update SA1402 to handle records and record structs

### LINKED ISSUE #3542: SA1402 does not fire for record classes
Repro:

```csharp
internal class Class1
{
}

internal class Class2 // SA1402 File may only contain a single type (expected)
{
}

internal record Class3 // no warning
{
}

internal record class Class4 // no warning
{
}
```

Expected behavior: SA1402 is reported for record classes, same as for non-record classes.

C# version: 10.0
StyleCopAnalyzers: 1.2.0-beta.406, 1.2.0-beta.435

### PR DESCRIPTION
**Pull request by bjornhellander**
Fixes #3542

### CODE REVIEW

**Comment by bjornhellander** [comment]
Ready for review

**Comment by bjornhellander** [comment]
Note that I did not add any tests for `record class`, since there is no new syntax node for it. Let me know if tests are still expected.

**Comment by bjornhellander** [comment]
Can you take a look at this, @sharwell?

**Comment by bjornhellander** [comment]
@sharwell?

**Review by sharwell — CHANGES_REQUESTED** [review]
Marking request changes to get the updates mentioned in comments.

File: StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp10/MaintainabilityRules/SA1402CSharp10ForRecordStructUnitTests.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Tunnel Vision Laboratories, LLC. All Rights Reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#nullable disable
```
[the remark this answers is not in the document]
Comment (bjornhellander) [reply StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp10/MaintainabilityRules/SA1402CSharp10ForRecordStructUnitTests.cs]: Remove it

File: StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp10/MaintainabilityRules/SA1402CSharp10ForRecordStructUnitTests.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace StyleCop.Analyzers.Test.CSharp10.MaintainabilityRules
+{
+    using StyleCop.Analyzers.Test.MaintainabilityRules;
+
+    public class SA1402CSharp10ForRecordStructUnitTests : SA1402ForBlockDeclarationUnitTestsBase
```
[the remark this answers is not in the document]
Comment (bjornhellander) [reply StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp10/MaintainabilityRules/SA1402CSharp10ForRecordStructUnitTests.cs:8]: Added tests

**Review by sharwell — APPROVED** [review]

**Pull request merged**